controllers/image_generator.go

```diff
@@ -164,17 +169,8 @@ func (r *ImageGenerator) getCloudProviderFromInfra() string {
 	return strings.ToLower(string(infrastructure.Status.PlatformStatus.Type))
 }
 
-func (r *ImageGenerator) setupCloudProvider(cm *corev1.ConfigMap) error {
-	var provider string
-	if r.provider != "" && r.CMimageIDKey != "" {
-		return nil
-	}
-
-	if cm.Data["CLOUD_PROVIDER"] != "" {
-		provider = cm.Data["CLOUD_PROVIDER"]
-	} else {
-		provider = r.getCloudProviderFromInfra()
-	}
+func (r *ImageGenerator) setupCloudProvider() error {
+	provider := r.getCloudProviderFromInfra()
```

Comment:
It isn't obvious why all this code is being dropped from `setupCloudProvider()`.